concent_api/core/models.py

```diff
@@ -485,3 +493,478 @@ def clean(self):
             raise ValidationError({
                 'pending_response': 'PaymentInfo should be related with Pending Response'
             })
+
+
+class AbstractMessageModel(Model):
+
+    sig = BinaryField()
+    timestamp = IntegerField()
+    encrypted = BooleanField(default = False)
+
+    class Meta:
+        abstract = True
+
+
+class StoredComputeTaskDef(AbstractMessageModel):
+
+    # Defines items list from ComputeTaskDef
+    ITEMS = [
+        'task_id',
+        'subtask_id',
+        'deadline',
+        'src_code',
+        'extra_data',
+        'short_description',
+        'working_directory',
+        'performance',
+        'docker_images',
+    ]
+
+    task_id = CharField(max_length=MESSAGE_TASK_ID_MAX_LENGTH)
+    subtask_id = CharField(max_length=MESSAGE_TASK_ID_MAX_LENGTH)
+    deadline = DateTimeField(blank = True, null = True)
+    src_code = TextField()
+    extra_data = JSONField()
+    short_description = TextField()
+    working_directory = TextField()
+    performance = DecimalField(decimal_places=2, max_digits=3)
+    docker_images = JSONField()
+
+    if not len(ITEMS) == len(message.tasks.ComputeTaskDef().ITEMS.keys()):
+        raise ValidationError({
+            'ComputeTaskDef': 'StoredComputeTaskDef model has not same fields as golem_messages.message.tasks.ComputeTaskDef()'
+        })
+
+    for model_item, golem_message_item in zip(ITEMS, message.tasks.ComputeTaskDef().ITEMS.keys()):
+        if not model_item == golem_message_item:
+            raise ValidationError({
+                'ComputeTaskDef': 'StoredComputeTaskDef model has not same fields as golem_messages.message.tasks.ComputeTaskDef()'
+            })
+    '''
+    VALIDATION FOR task_id
+    VALIDATION FOR subtask_id
+    VALIDATION FOR deadline
+    VALIDATION FOR src_code
+    VALIDATION FOR extra_data
+    VALIDATION FOR short_description
+    VALIDATION FOR working_directory
+    VALIDATION FOR performance
+    VALIDATION FOR docker_images
+    '''
+
+
+class StoredTaskToCompute(AbstractMessageModel):
+
+    ITEMS = [
+        'requestor_id',
+        'requestor_public_key',
+        'requestor_ethereum_public_key',
+        'provider_id',
+        'provider_public_key',
+        'provider_ethereum_public_key',
+        'compute_task_def',
+        'package_hash',
+        'concent_enabled',
+        'price',
+        'header',
+        'sig',
+    ]
+
+    requestor_id = BinaryField()
+    requestor_public_key = BinaryField()
+    requestor_ethereum_public_key = BinaryField()
+    provider_id = BinaryField()
+    provider_public_key = BinaryField()
+    provider_ethereum_public_key = BinaryField()
+    compute_task_def = ForeignKey(StoredComputeTaskDef)
+    package_hash = CharField(max_length=HASH_LENGTH)
```

Comment:
You should include the prefix length here.

If it's already included in `HASH_LENGTH`, please make it a separate constant.